### PR TITLE
Created ad.py

### PR DESCRIPTION
**Pull request by vamshideveloper**
Module for Active directory
which will help us to manage AD through automation 
I will also update hvac/hvac/api/secrets_engines/__init__.py to support ad.py

### CODE REVIEW

File: hvac/api/secrets_engines/ad.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import logging
+
+from hvac import exceptions
+from hvac.api.vault_api_base import VaultApiBase
+
+DEFAULT_MOUNT_POINT = 'ad'
+
+
+class Ad(VaultApiBase):
+    """Google Cloud Secrets Engine (API).
```
Comment (jeffwecan) [inline hvac/api/secrets_engines/ad.py]: Missing an update here ;)

File: hvac/api/secrets_engines/ad.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    Reference: https://www.vaultproject.io/api/secret/ad/index.html
+    """
+
+    def configure(self, binddn="", bindpass="", url="", userdn==None, upndomain==None, ttl=0, max_ttl=0,
```
Comment (jeffwecan) [inline hvac/api/secrets_engines/ad.py]: Have one too many `=`'s in a couple spots:

`userdn==None, upndomain==None,` -> `userdn=None, upndomain=None,`

File: hvac/api/secrets_engines/ad.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+import json
+import logging
+
+from hvac import exceptions
+from hvac.api.vault_api_base import VaultApiBase
+
+DEFAULT_MOUNT_POINT = 'ad'
+
+
+class Ad(VaultApiBase):
```
Comment (jeffwecan) [inline hvac/api/secrets_engines/ad.py]: I'd rather name the class `ActiveDirectory` to make its purpose clearer.

**Comment by vamshideveloper** [comment]
Thanks for review, I will update the changes 
but i cannot edit the file

**Comment by jeffwecan** [comment]
It looks like your fork was deleted. You'll probably want to recover that fork in order to add more commits here...

**Pull request closed without merging**